sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+import { DeepMocked, createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames, getISODateOnlyString } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../test/helpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  mockDownloadFiles,
+  saveFakeStudent,
+  saveFakeStudentRestriction,
+  SFAS_ALL_RESTRICTIONS_FILENAME,
+  SFAS_LEGACY_RESTRICTION_FILENAME,
+} from "@sims/test-utils";
+import * as Client from "ssh2-sftp-client";
+import { Job } from "bull";
+import * as path from "path";
+import { SFASIntegrationScheduler } from "../sfas-integration.scheduler";
+import { In } from "typeorm";
+import { RestrictionCode } from "@sims/services";
+
+describe(describeProcessorRootTest(QueueNames.SFASIntegration), () => {
+  let app: INestApplication;
+  let processor: SFASIntegrationScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+  let sfasDownloadFolder: string;
+
+  beforeAll(async () => {
+    sfasDownloadFolder = path.join(__dirname, "sfas-files");
+    // Set the SFAS folder to the files mocks folders.
+    process.env.SFAS_RECEIVE_FOLDER = sfasDownloadFolder;
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(SFASIntegrationScheduler);
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    // Empty the student restrictions table for each test.
+    const studentRestrictionIds = await db.studentRestriction.find({
```

Comment:
Is this update not sufficient.

![image](https://github.com/bcgov/SIMS/assets/54600590/c23a0f27-8851-433d-ba9a-792e86119506)

Reply:
This won't work since I am also checking the count of the student restrictions inserted per test. If I don't delete the student restrictions and just update them to `inActive` status, then I cannot track the total number of student restrictions inserted per test.

Reply:
Thanks for the change.